arch/x86/core/irq_manage.c

```diff
@@ -99,3 +99,255 @@ void _arch_isr_direct_footer(int swap)
 	}
 }
 
+#if CONFIG_X86_DYNAMIC_IRQ_STUBS > 0
+
+/*
+ * z_interrupt_vectors_allocated[] bitfield is generated by the 'gen_idt' tool.
+ * It is initialized to identify which interrupts have been statically
+ * connected and which interrupts are available to be dynamically connected at
+ * run time, with a 1 bit indicating a free vector.  The variable itself is
+ * defined in the linker file.
+ */
+extern unsigned int z_interrupt_vectors_allocated[];
+
+struct dyn_irq_info {
+	/** IRQ handler */
+	void (*handler)(void *param);
+	/** Parameter to pass to the handler */
+	void *param;
+};
+
+/*
+ * Instead of creating a large sparse table mapping all possible IDT vectors
+ * to dyn_irq_info, the dynamic stubs push a "stub id" onto the stack
+ * which is used by common_dynamic_handler() to fetch the appropriate
+ * information out of this much smaller table
+ */
+static struct dyn_irq_info dyn_irq_list[CONFIG_X86_DYNAMIC_IRQ_STUBS];
+static unsigned int next_irq_stub;
+
+/* Memory address pointing to where in ROM the code for the dynamic stubs are.
+ * Linker symbol.
+ */
+extern char z_dynamic_stubs_begin[];
+
+#ifndef CONFIG_X86_FIXED_IRQ_MAPPING
+/**
+ * @brief Allocate a free interrupt vector given <priority>
+ *
+ * This routine scans the z_interrupt_vectors_allocated[] array for a free vector
+ * that satisfies the specified <priority>.
+ *
+ * This routine assumes that the relationship between interrupt priority and
+ * interrupt vector is :
+ *
+ *      priority = (vector / 16) - 2;
+ *
+ * Vectors 0 to 31 are reserved for CPU exceptions and do NOT fall under
+ * the priority scheme. The first vector used for priority level 0 will be 32.
+ *
+ * Each interrupt priority level  contains 16 vectors, and the prioritization
+ * of interrupts within a priority  level is determined by the vector number;
+ * the higher the vector number, the higher the priority within that priority
+ * level.
```

Comment:
Don't think this sentence is true.  The APIC will only trigger an interrupt on priority change, so there's no notion of priority "within that priority level".

Reply:
Removed. This was some very old verbiage from the Viper/WR days.